feat: Add modular dashboard with TemplUI components

Parvaggarwal01 opened the pull request:
Add comprehensive dashboard with TemplUI components, local card component, and responsive layout.

Parvaggarwal01 committed:
feat: Add modular dashboard with TemplUI components

Parvaggarwal01 committed:
Merge branch 'templui'
 commit.

ananyatimalsina commented:
This is good to see. However there are some issues with the commit as is:

- Please refrain from including unrelated files with your PR (.env in this case)
- The Dashboard page is too modular in places where it doesn't make any sense to be. The components that are used only once should directly go into the main component.

Parvaggarwal01 committed:
simplify dashboard by inlining single-use components

Parvaggarwal01 committed:
Merge remote-tracking branch 'upstream/main'

Parvaggarwal01 committed:
Merge remote-tracking branch 'upstream/main'

Parvaggarwal01 commented:
Fixed over-modularization:

- Refactored the dashboard to inline single-use components directly into the main Dashboard template

- Removed unnecessary abstractions like DashboardHeader(), StatsGrid(), ActiveEventsSection(), etc.

ananyatimalsina commented:
The Recent Activity seems to be bugged. As well as that, the table for Active Voting Events is inconsistent when compared the the modemnagement one. Also you might consider redirecting to the Events page when clicking New Event, though I am still unsure about that being the default action, as a regular User probably wouldn't interact with creating Events nearly enough to justify if being there.
